### PR TITLE
Add templates for account creation

### PR DESCRIPTION
**Pull request by flozia**
This PR adds updates the templates for the In-App Account creation.

### CODE REVIEW

File: src/authenticationinapp/authenticationinapp.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ void AuthenticationInApp::signIn() {
   m_listener->signIn();
 }
 
-const QString& AuthenticationInApp::emailAddress() const {
+const QString AuthenticationInApp::emailAddress() const {
```
Comment (bakulf) [inline src/authenticationinapp/authenticationinapp.cpp:93]: Why?

File: src/authenticationinapp/authenticationinapp.h [hunk 2]
```diff
@@ -86,7 +86,11 @@ class AuthenticationInApp final : public QObject {
   // Sign In/Up.
   Q_INVOKABLE void signIn();
   Q_INVOKABLE void signUp();
-  Q_INVOKABLE const QString& emailAddress() const;
+  Q_INVOKABLE const QString emailAddress() const;
```
Comment (bakulf) [inline src/authenticationinapp/authenticationinapp.h:89]: why this?

File: src/authenticationinapp/authenticationinapp.h [hunk 3]
```diff
@@ -86,7 +86,11 @@ class AuthenticationInApp final : public QObject {
...
+
+  Q_INVOKABLE int getVerificationCodeLength() {
```
Comment (bakulf) [inline src/authenticationinapp/authenticationinapp.h:91]: expose it as a property:

`Q_PROPERTY(int verificationCodeLength READ verificationCodeLength CONSTANT)`

File: src/authenticationinapp/authenticationinapp.h [hunk 4]
```diff
@@ -135,6 +139,8 @@ class AuthenticationInApp final : public QObject {
  private:
   State m_state = StateInitializing;
 
+  int m_verificationCodeLength = 6;
```
Comment (bakulf) [inline src/authenticationinapp/authenticationinapp.h:142]: This seems super static, right? Don't make it a variable. Do this:
1. add Q_PROPERTY as written above
2. in the .h write only: `static int verificationCodeLength() { return 6; }`

File: src/ui/authenticationInApp/ViewAuthenticationVerificationSessionByEmailNeeded.qml [hunk 5]
```diff
@@ -27,40 +30,84 @@ Item {
...
+                }
+            ]
+        }
 
-        anchors.top: msg.bottom
-        anchors.bottomMargin: 24
-        width: parent.width
+        VPNButton {
+            id: createAccountButton
+            enabled: codeInput.text && codeInput.text.length === VPNAuthInApp.getVerificationCodeLength()
```
Comment (bakulf) [inline src/ui/authenticationInApp/ViewAuthenticationVerificationSessionByEmailNeeded.qml:75]: This becomes: `VPNAuthInApp.verificationCodeLength`

File: src/ui/authenticationInApp/ViewAuthenticationVerificationSessionByEmailNeeded.qml [hunk 6]
```diff
@@ -27,40 +30,84 @@ Item {
...
-    VPNButton {
-        anchors.top: codeButton.bottom
-        anchors.bottomMargin: 24
-        text: "Send new code" // TODO
-        anchors.horizontalCenterOffset: 0
-        anchors.horizontalCenter: parent.horizontalCenter
-        radius: 5
-        onClicked: VPNAuthInApp.resendVerificationSessionCodeEmail();
+    Component.onCompleted: {
+        console.log("SESSION VERIFICATION BY EMAIL");
```
Comment (bakulf) [inline src/ui/authenticationInApp/ViewAuthenticationVerificationSessionByEmailNeeded.qml:110]: When ready, we can remove these messages.

**Review by bakulf — CHANGES_REQUESTED** [review]

File: src/ui/authenticationInApp/ViewAuthenticationVerificationSessionByEmailNeeded.qml (hunk 6 above)
Comment (flozia) [reply src/ui/authenticationInApp/ViewAuthenticationVerificationSessionByEmailNeeded.qml]: Done!

File: src/authenticationinapp/authenticationinapp.h (hunk 3 above)
Comment (flozia) [reply src/authenticationinapp/authenticationinapp.h]: Makes sense, thanks!

File: src/authenticationinapp/authenticationinapp.cpp (hunk 1 above)
Comment (flozia) [reply src/authenticationinapp/authenticationinapp.cpp:93]: It seems like `VPNAuthInApp.emailAddress()` is `undefined` in QML when it’s returning a reference.

File: src/authenticationinapp/authenticationinapp.h (hunk 2 above)
Comment (flozia) [reply src/authenticationinapp/authenticationinapp.h:92]: See comment above.

File: src/authenticationinapp/authenticationinapp.cpp (hunk 1 above)
Comment (bakulf) [reply src/authenticationinapp/authenticationinapp.cpp:93]: Sorry, I change my mind, make it invokable in the header file:
`Q_INVOKABLE const QString& emailAddress() const  { return m_listener->emailAddress(); }`

Comment (bakulf) [reply src/authenticationinapp/authenticationinapp.cpp:93]: But keep the &

**Review by bakulf — CHANGES_REQUESTED** [review]

**Review by lesleyjanenorton — APPROVED** [review]

**Review by bakulf — CHANGES_REQUESTED** [review]
perfect.

**Review by bakulf — APPROVED** [review]

**Pull request merged**